metrics/web/lib/dashboard/presentation/state/project_metrics_notifier.dart

```diff
@@ -567,6 +596,33 @@ class ProjectMetricsNotifier extends ChangeNotifier {
     }
   }
 
+  /// Updates the [_currentPageParameters] with the given [projectFilter] and
+  /// [projectGroupId].
+  ///
+  /// If the given [projectFilter] or [projectGroupId] is `null`,
+  /// current values are used.
+  void _updateCurrentPageParameters({
+    String projectFilter,
+    String projectGroupId,
+  }) {
+    final updatedprojectFilter = projectFilter ?? _projectNameFilter;
+    final updatedProjectGroupId = projectGroupId ?? _selectedProjectGroup?.id;
+
+    final updatedPageParameters = _currentPageParameters.copyWith(
+      projectFilter: updatedprojectFilter,
+      projectGroupId: updatedProjectGroupId,
+    );
```

Comment:
🔨
Maybe there is no sense in using the `copyWith` method here since it adds more complications to this mechanism, like checking if parameters are null?